### PR TITLE
Add Project X-Ray flow description

### PR DESCRIPTION
**Pull request by rw1nkler**
This PR adds the description of the VPR routing graph creation process and diagram of the general flow for Project X-Ray.

This should be merged after https://github.com/SymbiFlow/symbiflow-docs/pull/215.
Because the main SymbiFlow docs repository requires the same `sphinxcontrig-images` extension to handle these changes properly.

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -17,20 +17,9 @@ The project includes;
...
---------
-
 .. toctree::
    :maxdepth: 2
 
    getting-started
+   prjxray/index
    development/development
-
-
```
Comment (mithro) [inline docs/source/index.rst]: Why it this bit removed?

File: docs/source/prjxray/vtr-rrgraph.rst [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+
+This section will describe the prjxray database contents with respect to
+the routing graph.
+
+Grid
+----
+
+Project X-Ray documents one or more **parts**.  Within a **part** is a **grid**.
+The **grid** is documented in ``tilegrid.json`` and can be accessed via the
+prjxray API via the ``prjxray.db.Database``.grid method.
```
Comment (litghost) [inline docs/source/prjxray/vtr-rrgraph.rst]: ".grid" should be fixed width?

File: docs/source/prjxray/vtr-rrgraph.rst [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+or more **sites**, zero or more **tile wires** and zero or more **pips**.
+**Pips** are programmable interconnect points, and connect two **tile
+wires** together in a programmatic fashion.
+
+A **tile** may also have a **bits** definition if the output bitstream
+configures this tile. A **bits** definition consists of a **block
+type**, a **base address**, the number of **frames** in the **base
+address column**, a **word offset**, and a number of **words**.
+
+A **tile** may also have a **bits** definition if the output bitstream
```
Comment (litghost) [inline docs/source/prjxray/vtr-rrgraph.rst]: Repeated paragraph

File: docs/source/prjxray/vtr-rrgraph.rst [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+configures this tile. A **bits** definition consists of a **block
+type**, a **base address**, the number of **frames** in the **base
+address column**, a **word offset**, and a number of **words**.
+
+Routing fabric
+--------------
+
++------------------+------------------+
```
Comment (litghost) [inline docs/source/prjxray/vtr-rrgraph.rst]: This table is missing is label

File: docs/source/prjxray/vtr-rrgraph.rst [hunk 5]
```diff
@@ -0,0 +1,408 @@
...
+|                 | - only 1 site pin                                         |
++-----------------+-----------------------------------------------------------+
+
+PIP
+^^^
+
++-----------------+-----------------------------------------------------------+
+| Property        | Valid choices                                             |
++=================+===========================================================+
+| Property        | Valid choices                                             |
```
Comment (litghost) [inline docs/source/prjxray/vtr-rrgraph.rst]: Row appears twice?

File: docs/source/prjxray/vtr-rrgraph.rst [hunk 6]
```diff
@@ -0,0 +1,408 @@
...
+Step (1) - Group tile wires into “nodes”
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+   The first step is to first re-form nodes that contain all the
+   directly connected **tiles wires** These nodes are *not* VPR
+   **nodes**, they are simply the collection of **tile wires** that are
+   already a net (electrically equivalent).
+
+Step (2) - Classify “nodes”
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (litghost) [inline docs/source/prjxray/vtr-rrgraph.rst]: Stray character, messes up formating?

File: docs/source/prjxray/vtr-rrgraph.rst [hunk 7]
```diff
@@ -0,0 +1,408 @@
...
+`make_tracks <https://github.com/SymbiFlow/symbiflow-arch-defs/blob/master/utils/lib/rr_graph/tracks.py>`_.
+make_tracks takes a point bag containing all of the source and sink grid
+locations for a particular channel. It returns straight lines such that
+all sources and sink grid locations can route on to or off of the
+channel.
+
+Point Bag to CHANX / CHANY decomposition
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. figure:: ../images/prjxray/vtr-rrgraph.png
```
Comment (litghost) [inline docs/source/prjxray/vtr-rrgraph.rst]: This figure spills in the ToC? https://symbiflow--1365.org.readthedocs.build/projects/arch-defs/en/1365/prjxray/vtr-rrgraph.html

File: docs/source/prjxray/vtr-rrgraph.rst (hunk 5 above)
Comment (tmichalak) [reply docs/source/prjxray/vtr-rrgraph.rst]: Fixed.

File: docs/source/prjxray/vtr-rrgraph.rst (hunk 3 above)
Comment (tmichalak) [reply docs/source/prjxray/vtr-rrgraph.rst]: Fixed.

File: docs/source/prjxray/vtr-rrgraph.rst (hunk 6 above)
Comment (tmichalak) [reply docs/source/prjxray/vtr-rrgraph.rst]: Fixed.

File: docs/source/prjxray/vtr-rrgraph.rst (hunk 4 above)
Comment (rw1nkler) [reply docs/source/prjxray/vtr-rrgraph.rst]: Fixed

File: docs/source/prjxray/vtr-rrgraph.rst (hunk 2 above)
Comment (rw1nkler) [reply docs/source/prjxray/vtr-rrgraph.rst]: Fixed

File: docs/source/index.rst (hunk 1 above)
Comment (rw1nkler) [reply docs/source/index.rst]: Restored

File: docs/source/prjxray/vtr-rrgraph.rst (hunk 7 above)
Comment (rw1nkler) [reply docs/source/prjxray/vtr-rrgraph.rst]: Now it should be fine

File: docs/source/prjxray/vtr-rrgraph.rst [hunk 8]
```diff
@@ -0,0 +1,404 @@
...
+   -  This captures direct inter-tile connections like carry chain
+      wires, BRAM data cascade wires, etc.
+
+-  NULL - A node that has either no source or no sink. This wires
+   typically occur near the edge of the grid.
+
+-  EDGES_TO_CHANNEL - A node that sources and sinks from a **site pin**
+   and connects via a **pip** to a CHANNEL
+
+.. figure:: ../images/prjxray/import-wire-class.svg
```
Comment (litghost) [inline docs/source/prjxray/vtr-rrgraph.rst]: This image not working?

Comment (rw1nkler) [reply docs/source/prjxray/vtr-rrgraph.rst]: why?

rtd-img

Comment (litghost) [reply docs/source/prjxray/vtr-rrgraph.rst]: It appears to be working now

**Comment by rw1nkler** [comment]
Is it ready to be merged?

**Pull request merged**